library/std/src/fs.rs

```diff
@@ -1406,11 +1406,11 @@ impl Permissions {
     /// On Unix-based platforms this checks if *any* of the owner, group or others
     /// write permission bits are set. It does not check if the current
     /// user is in the file's assigned group. It also does not check ACLs.
-    /// Therefore even if this returns true you may not be able to write to the
-    /// file, and vice versa. The [`PermissionsExt`] trait gives direct access
-    /// to the permission bits but also does not read ACLs. If you need to
-    /// accurately know whether or not a file is writable use the `access()`
-    /// function from libc.
+    /// Therefore the return value of this function cannot be relied upon
+    /// to predict whether attempts to read or write the file will actually succeed.
+    /// The [`PermissionsExt`] trait gives direct access to the permission bits but
+    /// also does not read ACLs. If you need to accurately know whether or not a file
+    /// is writable use the `access()` function from libc.
```

Comment:
```suggestion
    /// is writable use the `access()` function from libc.
```
I think it's work mentioning that using `access` can lead to time-of-check-time-of-use problems where the permissions get changed between the call to access and the actual access, and therefore the return value of access *also* cannot be relied on for some things. The only way to figure out whether you can open a file is to open it.

Reply:
Perhaps the final sentence should simply be removed lest the documentation for this function turns into a tutorial about interacting with a file system.

Reply:
That sounds good.

Reply:
I agree with removing that line.